lib/Controller/WopiController.php

```diff
@@ -996,7 +996,7 @@ private function generateSettingToken(string $userId): string {
 	private function generateSettings(string $userId, string $type): array {
 		$nextcloudUrl = $this->appConfig->getNextcloudUrl() ?: trim($this->urlGenerator->getAbsoluteURL(''), '/');
 		$uri = $nextcloudUrl . '/index.php/apps/richdocuments/wopi/settings' . '?type=' . $type . '&access_token=' . $this->generateSettingToken($userId) . '&fileId=' . '-1';
-		$etag = $this->settingsService->getFolderEtag($type);
+		$etag = $this->settingsService->getFolderEtag($type) . $this->settingsService->getPresentationFolderEtag($type, $userId);
```

Comment:
Add a condition for guest users: if the user is a guest, then do not generate getPresentationFolderEtag.

Reply:
Actually I think we should change the method to also allow null for the userId, and not set it to a string in 

https://github.com/Rash419/richdocuments/blame/template-preset/lib/Controller/WopiController.php#L142

Using user ids that are non existing for non-logged in users may lead to more problems on other parts.

Reply:
@juliusknorr make sense